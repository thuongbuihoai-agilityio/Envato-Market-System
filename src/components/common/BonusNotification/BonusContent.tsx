import { Box, MenuList, Text } from '@chakra-ui/react';
import { memo } from 'react';

export type TBonusContentProps = {
  discount: number;
};

const BonusContentMenuList = ({ discount }: TBonusContentProps) => (
  <MenuList
    bg="background.component.primary"
    position="relative"
    p={6}
    border="none"
    minW={320}
    maxW={330}
  >
    <Box position="relative" w="full" p={5} bg="primary.600" borderRadius="lg">
      <Text
        textAlign="center"
        as="h3"
        color="white"
        fontSize={28}
        fontWeight="bold"
      >
        {discount}% Bonus
      </Text>
      <Text
        textAlign="center"
        px={5}
        fontSize="lg"
        lineHeight={6}
        color="primary.900"
      >
        Create an Account and Get Bonus
      </Text>
      <Box
        position="absolute"
        borderRadius="lg"
        top="5px"
        left="10px"
        bg="primary.800"
        h={132}
        w="calc(100% - 20px)"
        zIndex={-1}
      />
    </Box>
  </MenuList>
);

const BonusContent = memo(BonusContentMenuList);
export default BonusContent;
