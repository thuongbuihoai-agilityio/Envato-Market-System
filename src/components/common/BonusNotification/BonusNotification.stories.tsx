import type { Meta, StoryObj } from '@storybook/react';
import { Box, theme, useColorModeValue } from '@chakra-ui/react';

// Components
import BonusNotification from '.';

const meta: Meta<typeof BonusNotification> = {
  title: 'Custom Components/BonusNotification',
  tags: ['autodocs'],
  component: BonusNotification,
  argTypes: {
    colorFill: {
      description: 'The color fill of the bonus notification icon',
    },
    discount: {
      description: 'The discount percentage for the bonus notification',
    },
    isExpired: {
      description: 'Determine whether the bonus notification is expired or not',
      defaultValue: false,
    },
    limitOfBonus: {
      description: 'Determine the current number of bonus notifications limit',
      defaultValue: 5,
    },
  },
  decorators: [
    (Story) => (
      <Box h="20vh" bg="background.component.primary">
        <Story />
      </Box>
    ),
  ],
  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;

type Story = StoryObj<typeof BonusNotification>;

export const Primary: Story = {
  args: {
    isExpired: false,
    limitOfBonus: 5,
  },
  render: function Render(props) {
    const colorFill = useColorModeValue(
      theme.colors.gray[800],
      theme.colors.white,
    );

    return <BonusNotification {...props} colorFill={colorFill} />;
  },
};
