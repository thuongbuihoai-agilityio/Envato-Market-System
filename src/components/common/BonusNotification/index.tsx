import { Menu, MenuButton } from '@chakra-ui/react';
import { memo } from 'react';

// Components
import { IconButton } from '../..';
import { Gift } from '../../Icons';
import BonusContent from './BonusContent';

// Constants
import {
  DEFAULT_DISCOUNT_PERCENTAGE,
  LIMIT_OF_BONUS,
} from '@app/constants/discount';

export type TBonusNotificationProps = {
  isExpired?: boolean;
  colorFill: string;
  discount?: number;
  limitOfBonus?: number;
};
const BonusNotification = ({
  isExpired = false,
  colorFill,
  discount = DEFAULT_DISCOUNT_PERCENTAGE,
  limitOfBonus = LIMIT_OF_BONUS,
}: TBonusNotificationProps) => (
  <Menu placement="bottom-end" offset={[0, 20]}>
    <MenuButton as="button">
      <IconButton hasNewNotification quantityNotification={limitOfBonus}>
        <Gift color={colorFill} />
      </IconButton>
    </MenuButton>
    {!isExpired && <BonusContent discount={discount} />}
  </Menu>
);

const BonusNotificationMemorized = memo(BonusNotification);
export default BonusNotificationMemorized;
