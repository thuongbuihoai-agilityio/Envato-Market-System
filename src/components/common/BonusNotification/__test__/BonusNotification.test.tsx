import { render } from '@testing-library/react';
import BonusNotification from '..';

const setup = () => render(<BonusNotification colorFill="mock" />);
describe('BonusNotification test cases', () => {
  it('should render correctly', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
