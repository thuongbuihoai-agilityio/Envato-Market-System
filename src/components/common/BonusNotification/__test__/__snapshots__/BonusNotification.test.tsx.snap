// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BonusNotification test cases should render correctly 1`] = `
<div>
  <button
    aria-controls="menu-list-:r1:"
    aria-expanded="false"
    aria-haspopup="menu"
    class="chakra-menu__menu-button"
    id="menu-button-:r1:"
  >
    <span
      class="css-xl71ch"
    >
      <div
        class="css-1ve4u7y"
        data-testid="icon-button-component"
      >
        <p
          class="chakra-text css-tlt4oo"
        >
          5
        </p>
        <div
          aria-label="Send email"
          class="chakra-button css-yt1s1n"
          data-testid="icon-button"
        >
          <svg
            fill="none"
            height="24"
            stroke="mock"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 11H20M4 11C2.89543 11 2 10.1046 2 9V8C2 6.89543 2.89543 6 4 6H20C21.1046 6 22 6.89543 22 8V9C22 10.1046 21.1046 11 20 11M4 11L4 20C4 21.1046 4.89543 22 6 22H18C19.1046 22 20 21.1046 20 20V11M12 6H16C17.1046 6 18 5.10457 18 4C18 2.89543 17.1046 2 16 2C13.7909 2 12 3.79086 12 6ZM12 6H8C6.89543 6 6 5.10457 6 4C6 2.89543 6.89543 2 8 2C10.2091 2 12 3.79086 12 6ZM12 6V22"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </svg>
        </div>
      </div>
    </span>
  </button>
  <div
    class="css-0"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-gqxg0i"
      id="menu-list-:r1:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="css-job24k"
      >
        <h3
          class="chakra-text css-owlh8f"
        >
          60
          % Bonus
        </h3>
        <p
          class="chakra-text css-3fpd34"
        >
          Create an Account and Get Bonus
        </p>
        <div
          class="css-n6nw9n"
        />
      </div>
    </div>
  </div>
</div>
`;
